git_theta/utils.py

```diff
@@ -302,3 +304,30 @@ def touch(path: str):
     """Update the access and modify time of `path`."""
     dt_epoch = datetime.datetime.now().timestamp()
     os.utime(path, (dt_epoch, dt_epoch))
+
+
+@contextlib.contextmanager
+def named_temporary_file(*args, **kwargs):
+    """A named temp file that is safe to use on windows.
+
+    When using this function to create a named tempfile, it is safe to call
+    operations like `.flush()` and `.close()` on the tempfile which is needed
+    on Windows. Like the normal tempfile context manager, the file is removed
+    automatically when you exit the `with` scope.
+    """
+    # We force these so remove them.
+    m = kwargs.pop("mode", None)
+    if m is not None:
+        raise RuntimeError(
+            f"'mode' argument should not be provided to 'named_temporary_file', got {m}."
+        )
+    d = kwargs.pop("delete", None)
+    if d is not None:
+        raise RuntimeError(
+            f"'delete' argument should not be provided to 'named_temporary_file', got {d}."
+        )
+    with tempfile.NamedTemporaryFile(*args, mode="w", delete=False, **kwargs) as f:
```

Comment:
Maybe I'm misunderstanding, but couldn't you call `named_temporary_file` with positional args that would then cause a argument disagreement? Like, the first positional arg to NamedTemporaryFile is mode, so if I call named_temporary_file("r") I think this line would be providing two different values for the mode argument?

Reply:
Removed the `*args` part, so should still be flexible without that issue.